bool next_move
---
bool ok
